examples/js/loaders/GLTFLoader.js

```diff
@@ -12,368 +16,1836 @@ THREE.GLTFLoader.prototype = {
+GLTFInterpolator.prototype.interp = function( t ) {
+
+	if ( t == this.keys[ 0 ] ) {
+
+		if ( this.isRot ) {
+
+			this.quat3.set( this.values[ 0 ], this.values[ 1 ], this.values[ 2 ], this.values[ 3 ] );
+
+		} else {
+
+			this.vec3.set( this.values[ 0 ], this.values[ 1 ], this.values[ 2 ] );
 
 		}
 
-		// accessors
+	} else if ( t < this.keys[ 0 ] ) {
 
-		var COMPONENT_TYPES = {
-			5120: Int8Array,
-			5121: Uint8Array,
-			5122: Int16Array,
-			5123: Uint16Array,
-			5125: Uint32Array,
-			5126: Float32Array,
-		};
+		if ( this.isRot ) {
 
-		var TYPE_SIZES = {
-			'SCALAR': 1, 'VEC2': 2, 'VEC3': 3, 'VEC4': 4,
-			'MAT2': 4, 'MAT3': 9, 'MAT4': 16
-		};
+			this.quat1.set( this.originalValue.x,
+					this.originalValue.y,
+					this.originalValue.z,
+					this.originalValue.w );
+			this.quat2.set( this.values[ 0 ],
+					this.values[ 1 ],
+					this.values[ 2 ],
+					this.values[ 3 ] );
+			THREE.Quaternion.slerp( this.quat1, this.quat2, this.quat3, t / this.keys[ 0 ] );
 
-		var accessors = json.accessors;
+		} else {
 
-		for ( var accessorId in accessors ) {
+			this.vec3.set( this.originalValue.x,
+					this.originalValue.y,
+					this.originalValue.z );
+			this.vec2.set( this.values[ 0 ],
+					this.values[ 1 ],
+					this.values[ 2 ] );
 
-			var accessor = accessors[ accessorId ];
+			this.vec3.lerp( this.vec2, t / this.keys[ 0 ] );
 
-			var arraybuffer = library.bufferViews[ accessor.bufferView ];
-			var itemSize = TYPE_SIZES[ accessor.type ];
-			var TypedArray = COMPONENT_TYPES[ accessor.componentType ];
+		}
 
-			var array = new TypedArray( arraybuffer, accessor.byteOffset, accessor.count * itemSize );
+	} else if ( t >= this.keys[ this.count - 1 ] ) {
+
+		if ( this.isRot ) {
+
+			this.quat3.set( this.values[ ( this.count - 1 ) * 4 ],
+					this.values[ ( this.count - 1 ) * 4 + 1 ],
+					this.values[ ( this.count - 1 ) * 4 + 2 ],
+					this.values[ ( this.count - 1 ) * 4 + 3 ] );
 
-			library.accessors[ accessorId ] = new THREE.BufferAttribute( array, itemSize );
+		} else {
+
+			this.vec3.set( this.values[ ( this.count - 1 ) * 3 ],
+					this.values[ ( this.count - 1 ) * 3 + 1 ],
+					this.values[ ( this.count - 1 ) * 3 + 2 ] );
 
 		}
 
-		// textures
+	} else {
 
-		var FILTERS = {
-			9728: THREE.NearestFilter,
-			9729: THREE.LinearFilter,
-			9984: THREE.NearestMipMapNearestFilter,
-			9985: THREE.LinearMipMapNearestFilter,
-			9986: THREE.NearestMipMapLinearFilter,
-			9987: THREE.LinearMipMapLinearFilter
-		};
+		for ( var i = 0; i < this.count - 1; i ++ ) {
 
-		var WRAPPINGS = {
-			33071: THREE.ClampToEdgeWrapping,
-			33648: THREE.MirroredRepeatWrapping,
-			10497: THREE.RepeatWrapping
-		};
+			var key1 = this.keys[ i ];
+			var key2 = this.keys[ i + 1 ];
 
-		var textures = json.textures;
+			if ( t >= key1 && t <= key2 ) {
 
-		for ( var textureId in textures ) {
+				if ( this.isRot ) {
 
-			var texture = textures[ textureId ];
+					this.quat1.set( this.values[ i * 4 ],
```

Comment:
Good tip. I went through the loader and fixed this wherever I could in e0e7dbf9fefa8ee6a01e580ca0801fd2c9bf0f40.